### PR TITLE
Added banner functionality for newCourse

### PR DESCRIPTION
**Pull request by abalanovsky**
Added banner functionality for newCourse

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/108689551/89ee92b4-de32-41c1-b602-c4090e51d1a4

#1306

### CODE REVIEW

**Review by dmtrth25 — APPROVED** [review]

File: src/containers/add-course-banner/AddCourseBanner.styles.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const getStyles = (bannerUrl: string) => ({
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.styles.ts:1]: rename getStyles to styles and pass bannerUrl not to the whole styles object, but only to the container property

File: src/containers/add-course-banner/AddCourseBanner.styles.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  container: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundSize: 'cover',
+    borderRadius: '10px 10px 0px 0px',
+    px: { md: 7, sm: 3, xs: 3 },
+    py: { md: 6, sm: 4, xs: 4 },
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.styles.ts:9]: p: { xs: '4 3', md: '6 7' }

File: src/containers/add-course-banner/AddCourseBanner.styles.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    height: '216px',
+    backgroundImage: `linear-gradient(0deg, rgba(38, 50, 56, 0.15) 0%, rgba(38, 50, 56, 0.15) 100%), url(${bannerUrl})`,
+    '&:hover': {
+      backgroundImage: `linear-gradient(0deg, rgba(38, 50, 56, 0.40) 0%, rgba(38, 50, 56, 0.40) 100%), url(${bannerUrl})`
+    }
+  },
+  description: {
+    typography: 'subtitle2',
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.styles.ts:17]: typography: TypographyVariantEnum.Subtitle2

File: src/containers/add-course-banner/AddCourseBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import React, { useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Box, Typography } from '@mui/material'
+import { getStyles } from '~/containers/add-course-banner/AddCourseBanner.styles'
+import { validationData } from '~/containers/add-course-banner/AddCourseBanner.constants'
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.tsx:5]: organize imports: 
-libs, 
-containers ,contexts, services, hooks, components
-utils, routes, const, types, styles

File: src/containers/add-course-banner/AddCourseBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <>
+      <Box
+        className='container'
+        onClick={() => inputRef.current?.click()}
+        sx={styles.container}
+      >
+        <Box sx={styles.titleWithIcon}>
+          <Box alt='Add image icon' component='img' src={addImageIcon} />
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.tsx:55]: ComponentEnum for img

File: src/containers/add-course-banner/AddCourseBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  const { addFiles } = useUpload({
+    files: [],
+    emitter: handleBackgroundUpload,
+    validationData
+  })
+
+  return (
+    <>
+      <Box
+        className='container'
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.tsx:50]: why do you need this className?

File: src/containers/add-course-banner/AddCourseBanner.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        onClick={() => inputRef.current?.click()}
```
Comment (Tolik170) [inline src/containers/add-course-banner/AddCourseBanner.tsx:51]: better create this function outside and use it here

File: src/pages/create-course/CreateCourse.tsx [hunk 8]
```diff
@@ -10,12 +10,17 @@ import { authRoutes } from '~/router/constants/authRoutes'
 import { ButtonTypeEnum, ButtonVariantEnum } from '~/types'
 
 import { styles } from '~/pages/create-course/CreateCourse.styles'
+import AddCourseBanner from '~/containers/add-course-banner/AddCourseBanner'
+import React from 'react'
```
Comment (Tolik170) [inline src/pages/create-course/CreateCourse.tsx:14]: organize imports, and importing React is redundant

**Comment by TSlashDreamy** [comment]
Maybe we are we supposed to have some loading spinner component to gave our user feedback that his image is actually uploading? (It would be good in case if user have unstable or slow internet connection)

File: src/containers/add-course-banner/AddCourseBanner.styles.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundSize: 'cover',
+    borderRadius: '10px 10px 0px 0px',
+    px: { md: 7, sm: 3, xs: 3 },
+    py: { md: 6, sm: 4, xs: 4 },
+    height: '216px',
+    backgroundImage: `linear-gradient(0deg, rgba(38, 50, 56, 0.15) 0%, rgba(38, 50, 56, 0.15) 100%), url(${bannerUrl})`,
+    '&:hover': {
+      backgroundImage: `linear-gradient(0deg, rgba(38, 50, 56, 0.40) 0%, rgba(38, 50, 56, 0.40) 100%), url(${bannerUrl})`
```
Comment (TSlashDreamy) [inline src/containers/add-course-banner/AddCourseBanner.styles.ts:13]: can we add and use this colors in out `app.pallete.ts`?

File: src/containers/add-course-banner/AddCourseBanner.tsx (hunk 5 above)
Comment (TSlashDreamy) [inline src/containers/add-course-banner/AddCourseBanner.tsx:55]: maybe `alt` needs translation too 😅

File: src/containers/add-course-banner/AddCourseBanner.styles.ts (hunk 1 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.styles.ts]: Done

File: src/containers/add-course-banner/AddCourseBanner.styles.ts (hunk 2 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.styles.ts]: There should be no paddings here at all.
Thanks, i deleted them.

File: src/containers/add-course-banner/AddCourseBanner.styles.ts (hunk 3 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.styles.ts]: Thanks, done

File: src/containers/add-course-banner/AddCourseBanner.tsx (hunk 4 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.tsx:10]: Done

File: src/containers/add-course-banner/AddCourseBanner.tsx (hunk 6 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.tsx]: Don't need it, deleted.

File: src/containers/add-course-banner/AddCourseBanner.tsx (hunk 7 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.tsx]: Thanks, done

File: src/pages/create-course/CreateCourse.tsx (hunk 8 above)
Comment (abalanovsky) [reply src/pages/create-course/CreateCourse.tsx]: Done

**Comment by abalanovsky** [comment]
> Maybe we are supposed to have some loading spinner component to gave our user feedback that his image is actually uploading? (It would be good in case if user have unstable or slow internet connection)

I'm not sure, cause currently we uploading image directly to browser without making any backend requests. We can add loader in the scope of New Course backend integration.

ps: My laptop recorded video with some lags, so u can be confused, but actually image uploads without any delays

File: src/containers/add-course-banner/AddCourseBanner.styles.ts (hunk 9 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.styles.ts:13]: Gradients colors used only for this component, so i don't see real reasons for supplementing palette with them.
Maybe we'll add it later when these colors are used somewhere else

File: src/containers/add-course-banner/AddCourseBanner.tsx (hunk 5 above)
Comment (abalanovsky) [reply src/containers/add-course-banner/AddCourseBanner.tsx]: Thanks, done

**Review by TSlashDreamy — APPROVED** [review]

**Review by Tolik170 — APPROVED** [review]

**Review by OlyaKorchan — APPROVED** [review]

**Pull request merged**